testcases/gcov_setup.py

```diff
@@ -0,0 +1,153 @@
+
+
+    def runTest(self):
+        print("OS: %s" %self.util.distro_name())
+        print("\nInstalling the ditro src...")
+        src_path = self.util.get_distro_src('kernel', '/root/', "-bp")
+        src_path = "/root/kernel-5.14.0-324.el9/linux-5.14.0-324.el9.ppc64le"
+        print("src_path=%s" %src_path)
+        print("\nadding gcov_param....")
+        self.kernel_config(src_path)
+        print("Building the new kernel...")
+        self.build_and_boot(src_path)
+
+
+
+    def kernel_config(self, dest_path):
+        """
+        enabling the gcov parameter
+        """
+        if dest_path is None:
+
+            log.error("Please provide a valid path")
+            return ""
+        k_version = "".join(self.cv_HOST.host_run_command("uname -r"))
+        boot_conf_file = f"/boot/config-{k_version}"
+        src_conf_file = f"{dest_path}/.config"
+        self.cv_HOST.host_run_command(f"cp {boot_conf_file} {src_conf_file}")
+        temp_dir = tempfile.mkdtemp(prefix="gcov_boot_conf_")
+        self.cv_HOST.copy_files_from_host(temp_dir, src_conf_file)
+        print(os.listdir(temp_dir))
+        config_file = "".join(os.listdir(temp_dir))
+        config_path = os.path.join(temp_dir, config_file)
+        print("config_path=%s" %config_path)
+        self.add_gcov_param(config_path)
+        self.cv_HOST.host_run_command("copying back the .config file")
+        self.cv_HOST.copy_test_file_to_host(config_path, temp_dir, "/tmp/")
+        self.cv_HOST.host_run_command("ls /tmp/")
+
+
+    def add_gcov_param(self, conf_file):
+        """
+        add the gcove kernel parameter
+        """
+        k_config_params = ["CONFIG_GCOV_KERNEL=Y", "CONFIG_ARCH_HAS_GCOV_PROFILE_ALL=Y"]
+        k_config_params.extend(["CONFIG_GCOV_PROFILE_ALL=Y", "CONFIG_GCOV_PROFILE_FTRACE=Y", "CONFIG_DEBUG_INFO_BTF=N"])
+        with open(conf_file, 'a+') as f:
+            file_content = f.read().splitlines()
+            for param in k_config_params:
+                count = 0
+                for line in file_content:
+                    print("\n\n checking param in file")
+                    if line.startswith('#') or line.strip() == "":
+                        continue
+                    elif line.split("=")[0] == param.split("=")[0]:
+                        print("param=%s found hence writting the param to file" %param)
+                        f.write("\n")
+                        f.write(line.replace(line, param))
+                        count += 1
+                        time.sleep(10)
+                print("count_value=%s, param=%s" %(count, param))
+                if not count:
+                    print("%s parameter_not_found, hence writing to .config file" %param)
+                    f.write(f'\n{param}')
+                    time.sleep(10)
+            cmd = f"cat {conf_file} | grep -i GCOV"
+            print("checking the config file after write: ")
+            print("cmd=%s" %cmd)
+            os.system(cmd)
```

Comment:
please cancel the test case here as without this param enable no point to go to the next step